postrprocess_visual/postprocess.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np
+import torch
+from torch import nn
+from torch.nn import functional as F
+
+
+class CenternetPostprocess(nn.Module):
+    detection_num = 100
+    """
+    Layer with input of CenterNet output.The layer output is tensor [batch_size, k, 6]
+    where 'k 'is max object number, '6' : (class_id, score, x_min, y_min, x_max, y_max)
+    """
+
+    def __init__(self, n_classes=80, width=320, height=320, down_ratio=4):
+        super().__init__()
+        self._n_classes = n_classes
+        self._down_ratio = down_ratio
+        self._width = width
+        self._height = height
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
+
+    def forward(self, y_pred):
+        hm = y_pred[:, :self._n_classes, :, :]
+        coors = y_pred[:, self._n_classes: self._n_classes + 4, :, :]
+        return self._ctdet_decode(hm, coors, output_stride=self._down_ratio, k=self.detection_num)
+
+    def _nms(self, hm, kernel=3):
+        hmax = F.max_pool2d(hm, kernel, stride=1, padding=kernel // 2)
+        keep = torch.eq(hmax, hm).type(torch.float32).to(self.device)
+        hm = hm.type(torch.float32)
+        return hm * keep
+
+    def _ctdet_decode(self, hm, coors, output_stride, k):
+        """
+        :param hm: class heatmaps [batch_size, rows, cols, class_number]
+        :param coors : width, height heatmaps [batch_size, rows, cols, 2] - 1st for width, second for height
+        :param output_stride: ratio between detector input and output image sizes
+        :param k: maximum number od detections on single image
+        :return: set of [class_id, score, x_min, y_min, x_max, y_max]
+        """
```

Comment:
What does this function do? Add a clear short description for it or think about better function name. Return type is not a Set, it's a Tensor

Reply:
done